crypto/dh/dh_key.c

```diff
@@ -230,3 +230,39 @@ static int dh_finish(DH *dh)
     BN_MONT_CTX_free(dh->method_mont_p);
     return 1;
 }
+
+int DH_buf2key(DH *key, const unsigned char *buf, size_t len)
+{
+    BIGNUM *pubkey;
+
+    pubkey = BN_bin2bn(buf, len, NULL);
+    if (!pubkey)
+        goto err;
+    if (BN_is_zero(pubkey))
+        goto err;
+    if (!DH_set0_key(key, pubkey, NULL))
+        goto err;
+    return 1;
+err:
+    DHerr(DH_F_BUF2KEY, ERR_R_INTERNAL_ERROR);
+    BN_free(pubkey);
+    return 0;
+}
+
+size_t DH_key2buf(const DH *dh, unsigned char **pbuf)
+{
+    const BIGNUM *pubkey;
+
+    if (!dh)
```

Comment:
Do not check for NULL parameters, that's a programming error.

Reply:
I think it's still an open question as to whether we should have explicit NULL checks or not. Opinions differ within the OMC I think.

Reply:
By "explicit NULL check" I mean "== NULL" instead of, say, using "!"   You're commenting on checking for NULL parameters which is different.  (I know you know, but others might not know, ya know?)  If keeping the NULL param check, probably an error needs to be pushed on the stack.

Reply:
Right...different meanings for what  "explicit NULL check" means. As you say I am commenting on whether we should check for NULL parameters or not.

Reply:
Removed this NULL parameter check